storey/flow.py

```diff
@@ -74,10 +76,25 @@ def __init__(
             self.name = type(self).__name__
 
         self._closeables = []
+        self._selected_outlet: Optional[list[str]] = None
+        self._create_name_to_outlet = True
 
     def _init(self):
         self._termination_received = 0
         self._termination_result = None
+        self._name_to_outlet = {}
+        if self._method_is_overridden("select_outlets", Flow) and self._create_name_to_outlet:
+            self._init_name_to_outlet()
+
+    def _init_name_to_outlet(self):
+        for outlet in self._outlets:
+            if outlet.name in self._name_to_outlet:
+                raise ValueError(f"Ambiguous outlet name '{outlet.name}' in Choice step")
```

Comment:
What Choice step? :)

```suggestion
                raise ValueError(f"Ambiguous outlet name '{outlet.name}' in step '{self.name}'")
```